### ISSUE TITLE
MacOS packaging requires additional permissions

### ISSUE THREAD

**Issue by droidmonkey** [keepassxreboot/keepassxc#2455]
## Expected Behavior

When ```make package``` is issued on MacOS by a non-administrator user, a DMG file would be produced without requiring an additional permissions.

## Current Behavior

The proxy executable is attempted to be copied to /usr/local/bin during the packaging process.

```
Run CPack packaging tool...
CPack: Create package using DragNDrop
CPack: Install projects
CPack: - Run preinstall target for: KeePassXC
CPack: - Install project: KeePassXC
CPack: -   Install component: Runtime
CMake Error at /Users/kpxc/keepassxc/build/src/proxy/cmake_install.cmake:44 (file):
  file INSTALL cannot copy file
  "/Users/kpxc/keepassxc/build/src/proxy/keepassxc-proxy" to
  "/usr/local/bin/keepassxc-proxy".
Call Stack (most recent call first):
  /Users/kpxc/keepassxc/build/src/cmake_install.cmake:42 (include)
  /Users/kpxc/keepassxc/build/cmake_install.cmake:37 (include)

error: /Library/Developer/CommandLineTools/usr/bin/strip: file: /usr/local/bin/keepassxc-proxy is not writable (Permission denied)
CMake Error at /Users/kpxc/keepassxc/build/src/cli/cmake_install.cmake:44 (file):
  file INSTALL cannot copy file
  "/Users/kpxc/keepassxc/build/src/cli/keepassxc-cli" to
  "/usr/local/bin/keepassxc-cli".
Call Stack (most recent call first):
  /Users/kpxc/keepassxc/build/src/cmake_install.cmake:44 (include)
  /Users/kpxc/keepassxc/build/cmake_install.cmake:37 (include)

error: /Library/Developer/CommandLineTools/usr/bin/strip: file: /usr/local/bin/keepassxc-cli is not writable (Permission denied)
CPack Error: Error when generating package: KeePassXC
make: *** [package] Error 1
```

## Possible Solution

Dunno, there is a bunch of custom install steps for MacOS in the CMake scripts.

**Comment by phoerious** [comment]
It does not require sudo. It may require some MAC permissions (dunno).

**Comment by weslly** [comment]
It doesn't require sudo if you have writing permissions to /usr/local/bin

**Comment by droidmonkey** [comment]
Why are we writing to /usr/local/bin for packaging?

**Comment by varjolintu** [comment]
I assume it's because of this: https://github.com/keepassxreboot/keepassxc/blob/develop/CMakeLists.txt#L269
```
set(CLI_INSTALL_DIR "/usr/local/bin")
set(PROXY_INSTALL_DIR "/usr/local/bin")
```

Those are probably just forgot to change. I think this would do the trick:
```
set(CLI_INSTALL_DIR "${PROGNAME}.app/Contents/MacOS")
set(PROXY_INSTALL_DIR "${PROGNAME}.app/Contents/MacOS")
```

**Comment by droidmonkey** [comment]
This is also very awkward:

https://github.com/keepassxreboot/keepassxc/blob/develop/CMakeLists.txt#L257-L260

**Comment by droidmonkey** [comment]
Also, it appears cpack is putting in a ton of other apps in the DMG??

![osx_dmg](https://user-images.githubusercontent.com/2809491/47998258-a6220e00-e0cc-11e8-9d8a-ad034a799bf4.png)

**Comment by varjolintu** [comment]
@droidmonkey It's just a link to the folder, but still odd. Noticed this when I had to exclude the folder because it was driving my editor nuts. No idea why this happens or why it is even needed.

**Comment by droidmonkey** [comment]
Yes sorry, they are links

**Comment by droidmonkey** [comment]
This was corrected with #2054
